osu.Game/Database/EFToRealmMigrator.cs

```diff
@@ -91,10 +94,20 @@ private void migrateBeatmaps(DatabaseWriteUsage ef)
                 }
                 else
                 {
-                    using (var transaction = realm.BeginWrite())
+                    var transaction = realm.BeginWrite();
+                    int written = 0;
+
+                    try
                     {
                         foreach (var beatmapSet in existingBeatmapSets)
                         {
+                            if (++written % 1000 == 0)
```

Comment:
In general I'm not a fan of using increment operators inside of larger expressions because that forces me to stop and carefully consider the order of evaluation in my head. I'd personally generally refrain, and either use `+= 1` or split the increment to its own line which is unambiguous.

As an aside, I think that the trick with the prefix `++` (which I presume is there mostly not to commit the very first opened transaction where nothing has been written yet) is causing `written` to over-count the maps imported thus far on the very first run (i.e. the first time this logic will run is after 999 actual map imports, not a thousand, I believe). But that's not really very important.

Reply:
That is correct. Didn't see it as a huge issue.